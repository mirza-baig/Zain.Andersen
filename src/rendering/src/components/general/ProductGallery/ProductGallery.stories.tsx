// Global
import { Story, Meta } from '@storybook/react';
// Local
import ProductGallery, { ProductGalleryProps } from './ProductGallery';
import defaultData from './ProductGallery.mock-data';

export default {
  title: 'General/ProductGallery',
  component: ProductGallery,
} as Meta;
const Template: Story<ProductGalleryProps> = (props) => <ProductGallery {...props} />;
export const Default = Template.bind({});
Default.args = defaultData;
