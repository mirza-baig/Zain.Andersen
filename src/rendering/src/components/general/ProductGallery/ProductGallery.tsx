// Global
import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';
import { Field, withDatasourceCheck } from '@sitecore-jss/sitecore-jss-nextjs';
// Components
import { Component } from 'src/helpers/Component';
import { ComponentProps } from 'lib/types/component-props';
import { useRef, useState } from 'react';
import { SliderWrapper } from 'src/helpers/SliderWrapper';
import ImageWrapper from 'src/helpers/Media/ImageWrapper';
import { getBreakpoint, useCurrentScreenType } from 'lib/utils/get-screen-type';
import classNames from 'classnames';
import { Item } from '@sitecore-jss/sitecore-jss-nextjs';
import { SliderRefType, SliderType } from 'src/helpers/SliderWrapper/SliderWrapper';

export type ProductGalleryProps =
  Feature.EnterpriseWeb.RenewalByAndersen.Components.General.ProductGallery.ProductGallery &
    ComponentProps;
const ProductGallery = (props: ProductGalleryProps) => {
  const { currentScreenWidth } = useCurrentScreenType();
  const isMobile = currentScreenWidth <= getBreakpoint('md');

  const [selectedImageIndex, setselectedImageIndex] = useState(0);

  const sliderRef = useRef<SliderType>();

  const { fields } = props;

  if (!fields) {
    return <></>;
  }

  const sliderSettings = {
    enableNumberedPagination: true,
    rows: 3,
    slidesPerRow: 3,
    infinite: false,
    responsive: [
      {
        breakpoint: getBreakpoint('md'),
        settings: {
          rows: 1,
          slidesPerRow: 1,
          slidesToShow: 3.4,
        },
      },
    ],
  };

  const getSpacingClasses = (index: number): string => {
    type ColumnIndices = 0 | 1 | 2;

    const maxItemsPerRow = 3;

    const columnWiseClasses: Record<ColumnIndices, string> = {
      0: 'md:pr-[10px]',
      1: 'md:px-[5px]',
      2: 'md:pl-[10px]',
    };

    return columnWiseClasses[(index % maxItemsPerRow) as ColumnIndices];
  };

  return (
    <Component
      variant={currentScreenWidth <= getBreakpoint('md') ? 'full' : 'lg'}
      padding={currentScreenWidth <= getBreakpoint('md') && 'px-0'}
      dataComponent="general/productgallery"
      {...props}
    >
      <div className="col-span-12 md:col-span-6">
        <div className="relative">
          <ImageWrapper
            image={{
              value: {
                src: fields?.images[selectedImageIndex]?.url,
                width: isMobile ? currentScreenWidth : 592,
                height: isMobile ? 375 : 592,
                alt: fields?.images[selectedImageIndex]?.fields.Alt.value,
              },
            }}
            imageLayout="intrinsic"
          />
        </div>
      </div>
      <div className="col-span-12 md:col-span-6 md:-mb-s">
        <div
          className={classNames(
            isMobile && '[&_.numbered-pagination_.slick-prev]:left-m',
            'md:[&_.numbered-pagination_.slick-prev]:left-[calc(100%-87px)] [&_.numbered-pagination_.slick-next]:right-m md:[&_.numbered-pagination_.slick-next]:left-[98%] md:[&_.numbered-pagination_ul]:mx-m md:[&_.numbered-pagination_ul]:text-right'
          )}
        >
          <SliderWrapper sliderSettings={sliderSettings} sliderRef={sliderRef as SliderRefType}>
            {fields?.images.map(
              (
                image: Item & {
                  fields?: {
                    Alt: Field<string>;
                  };
                },
                index: number
              ) => (
                <div
                  key={index}
                  className={classNames(
                    'pb-[15px]',
                    isMobile && 'mx-m pr-xxs',
                    getSpacingClasses(index)
                  )}
                >
                  <div
                    className={classNames(
                      index === selectedImageIndex &&
                        'relative before:absolute before:top-0 before:left-0 before:z-10 before:h-full before:w-full before:border-[4px] before:border-primary'
                    )}
                    onClick={() => setselectedImageIndex(index)}
                  >
                    <ImageWrapper
                      key={index}
                      image={{
                        value: {
                          src: image.url,
                          width: isMobile ? 100 : 187,
                          height: isMobile ? 100 : 187,
                          alt: image?.fields?.Alt?.value,
                        },
                      }}
                      imageLayout="responsive"
                    />
                  </div>
                </div>
              )
            )}
          </SliderWrapper>
        </div>
      </div>
    </Component>
  );
};

export default withDatasourceCheck()<ProductGalleryProps>(ProductGallery);
