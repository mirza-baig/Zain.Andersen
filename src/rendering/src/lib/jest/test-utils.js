// Global
import { render, fireEvent, waitFor } from '@testing-library/react';
import { ComponentPropsContext } from '@sitecore-jss/sitecore-jss-nextjs';
import { FastSitecoreContext } from 'lib/overrides/FastSitecoreContext';

// Lib
import { createSampleComponentFactory } from 'lib/mocks/mock-placeholder';

export const eeSitecoreContext = {
  route: {
    name: 'Foo',
    placeholders: {},
  },
  context: {
    pageEditing: true,
    site: {
      name: 'JssTestWeb',
    },
    language: 'en',
  },
};

export const renderComponent = (
  Component,
  {
    componentProps = {},
    staticProps = {},
    componentFactory = createSampleComponentFactory(),
    sitecoreContext = {
      route: {
        name: 'Foo',
        placeholders: {},
      },
    },
  }
) => {
  const mockLayoutData = {
    sitecore: sitecoreContext,
  };

  return render(
    <ComponentPropsContext value={staticProps}>
      <FastSitecoreContext componentFactory={componentFactory} layoutData={mockLayoutData}>
        <Component {...componentProps} />
      </FastSitecoreContext>
    </ComponentPropsContext>
  );
};

export const snapshot = (Component, mockData) => {
  const component = renderComponent(Component, mockData);
  expect(component.container.firstChild).toMatchSnapshot();
  return component;
};

export const hasDataComponent = async (component, name) => {
  await waitFor(() => {
    const nodes = component.baseElement.querySelectorAll(`[data-component="${name}"]`);
    return expect(nodes.length).toBe(1);
  });
};

export const mockDateAs = (epochMs) => {
  if (typeof epochMs !== 'number') {
    console.error(
      'mockDateAs only accepts epoch values in milliseconds to prevent timezones from factoring in to Date creation.'
    );
  }

  jest.spyOn(global.Date, 'now').mockImplementation(() => new Date(epochMs).valueOf());
};

export const testFieldError = async (Component, mockData, { label, value, error }) => {
  const component = renderComponent(Component, mockData);
  const field = component.getByLabelText(label);
  fireEvent.change(field, { target: { value } });
  field.blur();
  await waitFor(() => {
    const err = component.queryByText(error);
    return expect(err).not.toBe(null);
  });
};
