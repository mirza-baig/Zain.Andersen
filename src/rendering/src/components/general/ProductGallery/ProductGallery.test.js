// Global
import { hasDataComponent, snapshot } from 'lib/jest/test-utils';
import 'lib/jest/match-media-mock';
// Local
import ProductGallery from './ProductGallery';
import defaultData from './ProductGallery.mock-data';

it('renders correctly', async () => {
  const component = snapshot(ProductGallery, { componentProps: defaultData });
  await hasDataComponent(component, 'general/productgallery');
});
