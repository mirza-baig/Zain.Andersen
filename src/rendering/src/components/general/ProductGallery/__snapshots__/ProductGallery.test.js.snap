// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  data-component="general/productgallery"
  id="id1695242607"
>
  <div
    class="my-8 lg:mx-auto md:max-w-screen-lg theme-white"
  >
    <div
      class="section-grid grid grid-cols-2 md:grid-cols-12 px-m gap-s md:gap-s "
    >
      <div
        class="col-span-12 md:col-span-6"
      >
        <div
          class="relative"
        >
          <div
            class="relative hidden md:block  "
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27592%27%20height=%27592%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="Image"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <div
            class="relative w-full md:hidden"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27592%27%20height=%27592%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="Image"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; object-position: center;"
              />
              <noscript />
            </span>
          </div>
        </div>
      </div>
      <div
        class="col-span-12 md:col-span-6 md:-mb-s"
      >
        <div
          class="md:[&_.numbered-pagination_.slick-prev]:left-[calc(100%-87px)] [&_.numbered-pagination_.slick-next]:right-m md:[&_.numbered-pagination_.slick-next]:left-[98%] md:[&_.numbered-pagination_ul]:mx-m md:[&_.numbered-pagination_ul]:text-right"
        >
          <div
            class="slick-slider numbered-pagination slick-initialized"
          >
            
            <div
              class="slick-list"
            >
              <div
                class="slick-track"
                style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
              >
                <div
                  aria-hidden="false"
                  class="slick-slide slick-active slick-current"
                  data-index="0"
                  style="outline: none; width: 0px;"
                  tabindex="-1"
                >
                  <div>
                    <div
                      class="pb-[15px] md:pr-[10px]"
                      style="width: 33.333333333333336%; display: inline-block;"
                      tabindex="-1"
                    >
                      <div
                        class="relative before:absolute before:top-0 before:left-0 before:z-10 before:h-full before:w-full before:border-[4px] before:border-primary"
                      >
                        <div
                          class="relative hidden md:block aspect-picture   "
                        >
                          <span
                            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                          >
                            <span
                              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                            />
                            <img
                              alt="Image"
                              data-nimg="responsive"
                              decoding="async"
                              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                            />
                            <noscript />
                          </span>
                        </div>
                        <div
                          class="relative w-full md:hidden"
                        >
                          <span
                            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                          >
                            <span
                              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                            />
                            <img
                              alt="Image"
                              data-nimg="responsive"
                              decoding="async"
                              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; object-position: center;"
                            />
                            <noscript />
                          </span>
                        </div>
                      </div>
                    </div>
                    <div
                      class="pb-[15px] md:px-[5px]"
                      style="width: 33.333333333333336%; display: inline-block;"
                      tabindex="-1"
                    >
                      <div
                        class=""
                      >
                        <div
                          class="relative hidden md:block aspect-picture   "
                        >
                          <span
                            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                          >
                            <span
                              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                            />
                            <img
                              alt=""
                              data-nimg="responsive"
                              decoding="async"
                              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                            />
                            <noscript />
                          </span>
                        </div>
                        <div
                          class="relative w-full md:hidden"
                        >
                          <span
                            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                          >
                            <span
                              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                            />
                            <img
                              alt=""
                              data-nimg="responsive"
                              decoding="async"
                              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; object-position: center;"
                            />
                            <noscript />
                          </span>
                        </div>
                      </div>
                    </div>
                    <div
                      class="pb-[15px] md:pl-[10px]"
                      style="width: 33.333333333333336%; display: inline-block;"
                      tabindex="-1"
                    >
                      <div
                        class=""
                      >
                        <div
                          class="relative hidden md:block aspect-picture   "
                        >
                          <span
                            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                          >
                            <span
                              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                            />
                            <img
                              alt="jmage"
                              data-nimg="responsive"
                              decoding="async"
                              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                            />
                            <noscript />
                          </span>
                        </div>
                        <div
                          class="relative w-full md:hidden"
                        >
                          <span
                            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                          >
                            <span
                              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                            />
                            <img
                              alt="jmage"
                              data-nimg="responsive"
                              decoding="async"
                              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; object-position: center;"
                            />
                            <noscript />
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div>
                    <div
                      class="pb-[15px] md:pr-[10px]"
                      style="width: 33.333333333333336%; display: inline-block;"
                      tabindex="-1"
                    >
                      <div
                        class=""
                      >
                        <div
                          class="relative hidden md:block aspect-picture   "
                        >
                          <span
                            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                          >
                            <span
                              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                            />
                            <img
                              alt=""
                              data-nimg="responsive"
                              decoding="async"
                              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                            />
                            <noscript />
                          </span>
                        </div>
                        <div
                          class="relative w-full md:hidden"
                        >
                          <span
                            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                          >
                            <span
                              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                            />
                            <img
                              alt=""
                              data-nimg="responsive"
                              decoding="async"
                              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; object-position: center;"
                            />
                            <noscript />
                          </span>
                        </div>
                      </div>
                    </div>
                    <div
                      class="pb-[15px] md:px-[5px]"
                      style="width: 33.333333333333336%; display: inline-block;"
                      tabindex="-1"
                    >
                      <div
                        class=""
                      >
                        <div
                          class="relative hidden md:block aspect-picture   "
                        >
                          <span
                            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                          >
                            <span
                              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                            />
                            <img
                              alt=""
                              data-nimg="responsive"
                              decoding="async"
                              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                            />
                            <noscript />
                          </span>
                        </div>
                        <div
                          class="relative w-full md:hidden"
                        >
                          <span
                            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                          >
                            <span
                              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                            />
                            <img
                              alt=""
                              data-nimg="responsive"
                              decoding="async"
                              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; object-position: center;"
                            />
                            <noscript />
                          </span>
                        </div>
                      </div>
                    </div>
                    <div
                      class="pb-[15px] md:pl-[10px]"
                      style="width: 33.333333333333336%; display: inline-block;"
                      tabindex="-1"
                    >
                      <div
                        class=""
                      >
                        <div
                          class="relative hidden md:block aspect-picture   "
                        >
                          <span
                            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                          >
                            <span
                              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                            />
                            <img
                              alt=""
                              data-nimg="responsive"
                              decoding="async"
                              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                            />
                            <noscript />
                          </span>
                        </div>
                        <div
                          class="relative w-full md:hidden"
                        >
                          <span
                            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                          >
                            <span
                              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                            />
                            <img
                              alt=""
                              data-nimg="responsive"
                              decoding="async"
                              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; object-position: center;"
                            />
                            <noscript />
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div>
                    <div
                      class="pb-[15px] md:pr-[10px]"
                      style="width: 33.333333333333336%; display: inline-block;"
                      tabindex="-1"
                    >
                      <div
                        class=""
                      >
                        <div
                          class="relative hidden md:block aspect-picture   "
                        >
                          <span
                            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                          >
                            <span
                              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                            />
                            <img
                              alt="Image"
                              data-nimg="responsive"
                              decoding="async"
                              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                            />
                            <noscript />
                          </span>
                        </div>
                        <div
                          class="relative w-full md:hidden"
                        >
                          <span
                            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                          >
                            <span
                              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                            />
                            <img
                              alt="Image"
                              data-nimg="responsive"
                              decoding="async"
                              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; object-position: center;"
                            />
                            <noscript />
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            
            
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
